packages/utilities/src/CliUtils.ts

```diff
@@ -403,6 +402,108 @@ export class CliUtils {
         return prompt.question(message);
     }
 
+    /**
+     * Sleep for the specified number of miliseconds.
+     * @param timeInMs Number of miliseconds to sleep
+     *
+     * @example
+     *      // create a synchronous delay as follows:
+     *      await CliUtils.sleep(3000);
+     */
+    public static async sleep(timeInMs: number) {
+        return new Promise((resolve) => {
+            setTimeout(resolve, timeInMs);
+        });
+    }
+
+    /**
+     * Prompt the user with a question and wait for an answer,
+     * but only up to the specified timeout.
+     *
+     * @param questionText The text with which we will prompt the user.
+     *
+     * @param hideText Should we hide the text. True = display stars.
+     *                 False = display text. Default = false.
+     *
+     * @param secToWait The number of seconds that we will wait for an answer.
+     *                  If not supplied, the default is 30 seconds.
+     *
+     * @return A string containing the user's answer, or null if we timeout.
+     *
+     * @example
+     *      const answer = await CliUtils.promptWithTimeout("Type your answer here: ");
+     *      if (answer === null) {
+     *          // abort the operation that you wanted to perform
+     *      } else {
+     *          // use answer in some operation
+     *      }
+     */
+    public static async promptWithTimeout(
+        questionText: string,
+        hideText: boolean = false,
+        secToWait: number = 30
+    ): Promise<string> {
+
+        // readline provides our interface for terminal I/O
+        const readline = require("readline");
+        const ttyIo = readline.createInterface({
+            input: process.stdin,
+            output: process.stdout,
+            terminal: true,
+            prompt: questionText
+        });
+
+        // ask user the desired question and then asynchronously read answer
+        ttyIo.prompt();
+        let answerToReturn: string = null;
+        ttyIo.on("line", (answer: string) => {
+            answerToReturn = answer;
+            ttyIo.close();
+        }).on("close", () => {
+            if (hideText) {
+                // The user's Enter key was echoed as a '*', so now output a newline
```

Comment:
Is the Enter key echoed as a `*`? If so, should we disable that behavior for the Enter key in the `_writeToOutput` method below?

When entering a password, if its short (like mainframe passwords which are limited to 8 chars), I pay attention to the number of `*` characters. If I saw 9 instead of 8, I could be confused and think I had entered my password wrong. Maybe this is just me being weird, not sure 😄

Reply:
I believe there is a security concern if the masking of your password matches your password length.
So, it should be fine if we output 9 (or 10) `*` even if there is a restriction.

Reply:
I originally tried to test for the Enter key value within the _writeToOutput method. I was unable to get the test to work. Perhaps I did something wrong, but after enough attempts, I just gave up.

Reply:
I went back to this problem again. A different but very simple approach prevents a star from being displayed when the Enter key is typed.